backend/routers/voice.py

```
@@ -0,0 +1,321 @@
+logger = logging.getLogger(__name__)
+
+router = APIRouter()
+
+# Directory for storing audio files
+AUDIO_STORAGE_DIR = os.path.join(os.path.dirname(__file__), "..", "data", "audio_recordings")
+os.makedirs(AUDIO_STORAGE_DIR, exist_ok=True)
+
+
+@router.post("/api/voice/transcribe", response_model=VoiceTranscriptionResponse)
+async def transcribe_voice(
+    audio_file: UploadFile = File(..., description="Audio file (WAV, MP3, FLAC, etc.)"),
```

Ruff (0.15.1)

[warning] Do not perform function call `File` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)